src/components/atoms/InputField/InputField.scss

```diff
@@ -0,0 +1,51 @@
+@import "scss/constants";
+
+$icon-width: 2.5rem;
+
+.input-field {
+  position: relative;
+
+  &--icon-start {
+    .input-field__input {
+      padding-left: $icon-width;
+    }
+  }
+
+  &--icon-end {
+    .input-field__input {
+      padding-right: $icon-width;
+    }
+  }
+
+  &__input {
+    height: 100%;
+
+    &:focus + .input-field__icon > * {
+      opacity: 0.6;
+    }
+  }
+
+  &__icon {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    position: absolute;
+    top: 0;
+    height: 100%;
+    width: $icon-width;
+    z-index: z(input-icon);
+
+    &--start {
+      left: 0;
+    }
+
+    &--end {
+      right: 0;
+    }
```

Comment:
I'm not sure that would work. 
The logic is the following: if input-field has start/end icon prop, it achieves `input-field--icon-start` / `input-field--icon-end` className accordingly.

```scss
.input-field {
  &--icon-start {
    .input-field__icon {
      left: 0;
    }
  }

  &--icon-end {
    .input-field__icon {
      right: 0;
    }
  }
}
```

As we have the ability to add both icons (start and end), it means that parent div will have both classes, and the property `left: 0` will be ignored, as the icon can't be on the left and right sides simultaneously and both icons will appear on the right side.

or did you mean it differently?

Reply:
Yup, that’s a fair point. The way we are using the start/end icons now aren’t really —variants anymore, they are more like 2 distinct __elements; so that’s why my method here wouldn’t work.

Perhaps we should change the __element part so that they distinctly reference start/end icons.